src/components/dropdowns/notifications.js

```diff
@@ -1,25 +1,47 @@
 import React, { Component } from 'react'
 import { FormattedMessage } from 'react-intl'
+import { connect } from 'react-redux'
 import { Link } from 'react-router-dom'
 
 import Notification from 'components/notification'
-import data from '../../data'
+
+import origin from '../../services/origin'
 
 class NotificationsDropdown extends Component {
   constructor(props) {
     super(props)
+
+    this.state = { notifications: [] }
+  }
+
+  async componentWillMount() {
+    try {
+      const notifications = await origin.notifications.all()
+
+      this.setState({ notifications })
+    } catch(e) {
+      console.error(e)
+    }
   }
 
   render() {
-    // randomly select from three examples
-    const exampleCounts = [4, 44, 444][Math.floor(Math.random() * 3)]
+    const { web3Account } = this.props
+    const { notifications } = this.state
+    const notificationsWithPerspective = notifications.map(n => {
+      const { sellerAddress } = n.resources.listing
+
+      return { ...n, perspective: web3Account === sellerAddress ? 'seller' : 'buyer' }
+    })
+    const filteredNotifications = notificationsWithPerspective.filter(n => n.status === 'unread')
     // avoid integers greater than two digits
-    const notificationCount = exampleCounts < 100 ? Number(exampleCounts).toLocaleString() : `${Number(99).toLocaleString()}+`
+    const notificationCount = filteredNotifications.length < 100 ?
+                              Number(filteredNotifications.length).toLocaleString() :
+                              `${Number(99).toLocaleString()}+`
```

Comment:
@jordajm I think I saw that react-intl has some number localization utilities. Is it appropriate to replace this native localization method?

Reply:
Cool, I'm not aware of a compelling reason to use [Intl.NumberFormat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/NumberFormat).